File: hycom/MSCPROGS/src/ObsCompare/plotargocmp.py
#!/local/python-2.5.1-login-gnu/bin/python
import pylab as p
import re
from numpy import *
import os
import glob
#import matplotlib
#matplotlib.use('Agg')

#for fname in ( 'profile_date20080619_id0006900494_pos-18.21Ex44.67N' ) :
#if (True) :
#for fname in os.listdir("./*N") :
for fname in glob.glob("./*N") :
   #fname = 'profile_date20080620_id0004900438_pos-72.57Ex26.84N'
   print fname
   subs=fname.split("_")
   subs2=subs[3].split("pos")
   pos=subs2[1]
   title =  subs[1] + " Position:" + subs2[1] + "Float ID: " + subs[2]


   depths=[];
   msaln=[];
   mtemp=[];
   osaln=[];
   otemp=[];
   file = open(fname)
   for line in file:
      subs=line.split()
      depths.append(-float(subs[0]))
      osaln.append (float(subs[1]))
      msaln.append (float(subs[2]))
      otemp.append (float(subs[3]))
      mtemp.append (float(subs[4]))
   file.close()
   msaln=array(msaln)
   osaln=array(osaln)
   mtemp=array(mtemp)
   otemp=array(otemp)
   depths=array(depths)

   I=where(msaln != float(-999))
   J=where(osaln != float(-999))
   #print prod(I[0].shape), prod(J[0].shape), prod(msaln.shape), prod(osaln.shape)
   if ( prod(I[0].shape) > 1 and prod(J[0].shape) > 1 ) :
      p.hold(True)
      leg=[]
      if ( prod(I[0].shape) > 1 ) :
         p.plot(msaln[I],depths[I], color='r', linewidth=2)
         leg.append("Model")
      if ( prod(J[0].shape) > 1 ) :
         p.plot(osaln[J],depths[J], color='b', linewidth=2)
         leg.append("Argo")
      p.grid(True)
      p.ylabel("Depths[m]")
      p.xlabel("Salinity[psu]")
      p.title(title)
      p.legend(leg)
      p.savefig("saltprofile_" + pos + ".png")
      print "---saltprofile_" + pos + ".png"
      p.close()

   I=where(mtemp != -999)
   J=where(otemp != -999)
   #print prod(I[0].shape), prod(J[0].shape), prod(mtemp.shape), prod(otemp.shape)
   if ( prod(I[0].shape) > 1 and prod(J[0].shape) > 1 ) :
      p.hold(True)
      leg=[]
      if ( prod(I[0].shape) > 1 ) :
         p.plot(mtemp[I],depths[I], color='r', linewidth=2)
         leg.append("Model")
      if ( prod(J[0].shape) > 1 ) :
         p.plot(otemp[J],depths[J], color='b', linewidth=2)
         leg.append("Argo")
      p.grid(True)
      p.ylabel("Depths[m]")
      p.xlabel("Temperature[C]")
      p.title(title)
      p.legend(leg)
      p.savefig("tempprofile_" + pos + ".png")
      print "---tempprofile_" + pos + ".png"
      p.close()

File: hycom/MSCPROGS/src/ObsCompare/Matlab/plotargocmp.m
function plotargocmp(indir)
%List files
files=dir([ indir 'profile_date*'])

for i=1:prod(size(files))
   fname=files(i).name;
   dat=load([ indir fname] );


   % Get date from profile
   tmp=fname;
   tmp=strrep(tmp,'profile_date','');
   idate=regexprep(tmp,'_.*','');
   pos=regexprep(tmp,'.*pos','');
   epos=regexprep(pos,'Ex.*','');
   npos=regexprep(pos,'.*Ex',''); npos=regexprep(npos,'N.*','');
   disp([ fname ' ' epos ' ' npos]);

   %plot salinity
   I=find(dat(:,2)~=-999);
   J=find(dat(:,3)~=-999);
   if (prod(size(I))>1 & prod(size(J))>1)
      figure(1); clf ; hold on ;
      H1=[]; H2=[];
      legtxt1=[]; legtxt2=[];
      if (prod(size(I))>0)
         H1=plot(dat(I,2),-dat(I,1),'LineWidth',2);
         legtxt1='Data ';
      end
      if (prod(size(J))>0)
         H2=plot(dat(J,3),-dat(J,1),'LineWidth',2,'Color','r');
         legtxt2='Model';
      end
      set(gca,'FontSize',14);
      set(gca,'FontWeight','bold');
      grid on;
      H=[H1 H2];
      legtxt=[legtxt1 ; legtxt2];
      legend(H,legtxt); %,'Location','Best');
      xlabel('Salinity[psu]');
      ylabel('Depth[m]');
      print('-dpng','-r150',[ 'saltprofile_' pos '.png'])
      %print('-depsc2','-r300',[ 'saltprofile_' pos '.eps'])
      close(1);
   end

   %plot temperature
   I=find(dat(:,4)~=-999);
   J=find(dat(:,5)~=-999);
   H1=[]; H2=[];
   legtxt1=[]; legtxt2=[];
   if (prod(size(I))>1 & prod(size(J))>1)
      figure(2); clf ; hold on ;
      if (prod(size(I))>0)
         H1=plot(dat(I,4),-dat(I,1),'LineWidth',2);
         legtxt1='Data ';
      end
      if (prod(size(J))>0)
         H2=plot(dat(J,5),-dat(J,1),'LineWidth',2,'Color','r');
         legtxt2='Model';
      end
      set(gca,'FontSize',14);
      set(gca,'FontWeight','bold');
      grid on;
      H=[H1 H2];
      legtxt=[legtxt1 ; legtxt2];
      legend(H,legtxt); %,'Location','Best');
      xlabel('Temperature[C]');
      ylabel('Depth[m]');
      print('-dpng','-r150',[ 'tempprofile_' pos '.png'])
      close(2);
   end
end
